pkg/storage/snapshot/snapshot.go

```diff
@@ -741,25 +787,11 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "In error state"))
 	}
 
-	if vmSnapshotDeleting(vmSnapshotCpy) {
-		// Enable the vmsnapshot to be deleted only in case the source is unlocked AND
-		// (the operation completed completed OR
-		// after waiting until the content is deleted if needed)
-		if canRemoveFinalizer && (!vmSnapshotProgressing(vmSnapshotCpy) || contentDeletedIfNeeded(vmSnapshotCpy, content)) {
-			controller.RemoveFinalizer(vmSnapshotCpy, vmSnapshotFinalizer)
-		}
-	} else {
-		// since no status subresource can update metadata and status
```

Comment:
Was that required to be atomic from other components perspective?

Reply:
Shouldn't be necessary snapshot controller is the only thing that cares